shared/rakefiles/xk_config.rake

```diff
@@ -15,19 +16,30 @@ end
 # (and a new Serviceaccount Key) whenever @gcp_creds_file changes. This is not
 # what we want, i.e. don't create a new Key/key file when @gcp_creds_file
 # changes because of :configure_kubectl.
-@serviceaccount_key_file = ENV["TF_VAR_serviceaccount_key"]
-task :configure_serviceaccount => [@gcp_creds_file, :configure_current_project] do
+@serviceaccount_key_file = "/project/live/#{ENV['ENV']}/secrets/kube-system/owner.json"
+task :configure_serviceaccount => [@gcp_creds_file] do
   # TODO: This command is duplicated from exekube's gcp-project-init (and
   # hardcodes 'projectowner' instead of $SA_NAME which is only defined in
   # gcp-project-init). If gcp-project-init becomes idempotent (GPII-2989,
   # https://github.com/exekube/exekube/issues/92), or if this 'keys create'
   # step moves somewhere else in exekube, call this command from that place
   # instead.
-  sh "
-    [ -f $TF_VAR_serviceaccount_key ] || \
-    gcloud iam service-accounts keys create $TF_VAR_serviceaccount_key \
-      --iam-account projectowner@$TF_VAR_project_id.iam.gserviceaccount.com
-  "
+  unless File.file?(@serviceaccount_key_file)
+    sh "
+      gcloud iam service-accounts keys create #{@serviceaccount_key_file} \
+        --iam-account projectowner@$TF_VAR_project_id.iam.gserviceaccount.com
+    "
+  end
+end
+
+@default_creds_file = "/root/.config/gcloud/application_default_credentials.json"
+task :configure_app_default_login => [@default_creds_file]
+rule @default_creds_file do
+  # This retrieves application-default credentials using interactive auth,
+  # only in case service account credentials are not present.
+  unless File.file?(@serviceaccount_key_file)
```

Comment:
What do you think of creating an empty `@default_creds_file` (or, perhaps better, a file containing a comment like `# Empty application default credentials file because there is a serviceaccount key file instead`) so this task doesn't run every time?

Or maybe some kind of message when we avoid creating an ADC because we already have an SA key? I feel like it could be easy to get mixed up and end up with weird behavior.

Maybe I'm worrying too much and/or prematurely?

Reply:
IMHO file present/not present is sufficient mechanism here (involving actual content in the check just makes it unnecessary complicated)

Reply:
I am with @stepanstipl here.
To avoid confusion which auth method is being used, I'll add some debug output.